p2p/syncer.go

```diff
@@ -809,17 +808,16 @@ func (s *Syncer) Close() error {
 }
 
 // NewSyncer creates a new syncer listening on the given address.
-func NewSyncer(addr string, genesisID types.BlockID, cm *chain.Manager, tp *txpool.Pool, store SyncerStore, pm PeerManager) (*Syncer, error) {
+func NewSyncer(addr string, genesisID types.BlockID, cm *chain.Manager, tp *txpool.Pool, store PeerStore, npm NewPeerManager) (*Syncer, error) {
```

Comment:
`npm`!! 😱

Reply:
I think the interfaces are a bit confused here. When you have a constructor that takes a function and then call that function on another argument to the constructor...that's usually a bad sign.

Let's step back and ask: does `PeerManager` need to be an interface at all? I think not; its behavior is always the same. The only thing we want to be dynamic is the `PeerStore`. So let's move `p2putil.DefaultPeerManager` into `p2p`, and just call it `PeerManager`. Then, remove the `store` parameter from `NewSyncer`; it doesn't need it. Now, our initialization flow looks like:

```go
ps := p2putil.NewEphemeralStore()
pm := p2p.NewPeerManager(ps)
s := p2p.NewSyncer(addr, genesisID, cm, tp, pm)
```